sdk/core/core-client/src/serviceClient.ts

```diff
@@ -535,3 +540,24 @@ function flattenResponse(
     ...fullResponse.parsedBody
   });
 }
+
+function getCredentialScopes(options: ServiceClientOptions): string | string[] | undefined {
+  if (options.credentialScopes) {
+    const scopes = options.credentialScopes;
+    return Array.isArray(scopes)
+      ? scopes.map((scope) => new URL(scope).toString())
```

Comment:
@xirzec I added this to check that we get a valid URL, do you think this is good or did you have something better in mind?

Reply:
might be overkill, but I think it's a good start